src/types/model.ts

```diff
@@ -124,18 +129,37 @@ export class Registry {
    * @returns a registry instance
    */
   public static async load(
-    context: Context,
-    network: string
+    storage: DBService,
+    network: string,
+    genesisBlockNumber: number
   ): Promise<Registry> {
-    const str = await context.storage.getStr(getOrdersStorageKey(network));
-    const lastNotifiedError = await context.storage
-      .getStr(LAST_NOTIFIED_ERROR_STORAGE_KEY)
-      .then((isoDate) => (isoDate ? new Date(isoDate) : null))
+    const str = await storage
+      .getDB()
+      .get(
+        getNetworkStorageKey(CONDITIONAL_ORDER_REGISTRY_STORAGE_KEY, network)
+      );
+    const lastNotifiedError = await storage
+      .getDB()
+      .get(getNetworkStorageKey(LAST_NOTIFIED_ERROR_STORAGE_KEY, network))
+      .then((isoDate: string | number | Date) =>
+        isoDate ? new Date(isoDate) : null
+      )
+      .catch(() => null);
+
+    const lastProcessedBlock = await storage
+      .getDB()
```

Comment:
What about getting the db instance only once and re-using it in all the following calls?

Reply:
Done